### ISSUE TITLE
Update Golang to 1.12

### ISSUE THREAD

**Issue by benpye** [Microsoft/azure-pipelines-image-generation#741]
**Tool infomation**
- Tool name: Golang
- Add or update? Update
- Desired version: 1.12

**Areas and ecosystems**

- [ ] Android
- [ ] C / C++
- [ ] Container tools
- [ ] Database servers or clients
- [ ] General purpose tools
- [X] Go
- [ ] Java
- [ ] Javascript, Node.js, and TypeScript
- [ ] iOS, macOS, and Apple-specific tools
- [ ] .NET Core or .NET Desktop
- [ ] PHP
- [ ] Python
- [ ] Ruby
- [ ] Rust
- [ ] Scala
- [ ] Testing and code coverage
- [ ] Xamarin
- [ ] Other

**Image impacted**
Add the appropriate label for the image or images that should be updated (e.g., mac, ubuntu, window).
Mac, Windows and Linux

**Comment by benpye** [comment]
I would like to build my project (https://github.com/benpye/wsl-ssh-pageant) with Azure DevOps Pipelines however it requires Go 1.12 for support of AF_UNIX sockets on Windows. It would be great if at least one of the Windows images could be updated to support 1.12, but really this affects all images.

**Comment by faraonc** [comment]
I get `go: cannot find GOROOT directory: /usr/local/go1.12` using the Ubuntu image.

**Comment by bryanmacfarlane** [comment]
@faraonc - can you log a separate issue with more details and reference this issue?

**Comment by DavidStaheli** [comment]
@faraonc also please be aware that although this issue was closed, we're still waiting on deployment of the changes.  That can be tracked in Releases here: https://github.com/Microsoft/azure-pipelines-image-generation/releases

**Comment by Naatan** [comment]
What's the hold up on this? Go 1.12 was released over a month ago, and this issue addressed soon after. Little ironic that an update for a platform that facilitates continuous delivery is taking so long.
